connectors/sync_job_runner.py

```diff
@@ -118,6 +118,12 @@ async def execute(self):
                 await self._sync_done(sync_status=JobStatus.COMPLETED)
                 return
 
+            if hasattr(self.data_provider, "features"):
+                logger.warning(f"{self.data_provider.name} already has 'features' set.")
+            else:
+                logger.debug(f"Setting 'features' for {self.data_provider.name}")
+                self.data_provider.features = self.connector.features
```

Comment:
Should we have a method `set_features` in `BaseDataSource`?

I'm about to introduce a `set_logger` method as well: https://github.com/elastic/connectors-python/pull/806#discussion_r1231472801

Reply:
Makes sense, it's more consistent 👍